tests/test_cli.py

```diff
@@ -1,20 +1,110 @@
+from io import BytesIO
+from unittest.mock import patch
+from zipfile import ZipFile
+
+from freezegun import freeze_time
+
 from patronload.cli import main
 
 
-def test_cli_no_options(caplog, runner):
+@patch("patronload.database.oracledb")
+def test_cli_log_configured_from_env(
+    mocked_oracledb,  # pylint: disable=W0613
+    caplog,
+    monkeypatch,
+    mocked_s3,  # pylint: disable=W0613
+    runner,
+):
+    monkeypatch.setenv("LOG_LEVEL", "debug")
     result = runner.invoke(main)
     assert result.exit_code == 0
-    assert "Logger 'root' configured with level=INFO" in caplog.text
-    assert "Patronload config settings loaded for environment: test" in caplog.text
-    assert "Running patronload process" in caplog.text
-    assert "Total time to complete process" in caplog.text
+    assert "Logger 'root' configured with level=DEBUG" in caplog.text
 
 
-def test_cli_log_configured_from_env(caplog, monkeypatch, runner):
-    monkeypatch.setenv("LOG_LEVEL", "debug")
+@freeze_time("2023-03-01 12:00:00")
+@patch("patronload.database.oracledb")
+def test_cli_success(
+    mocked_oracledb,  # pylint: disable=W0613
+    caplog,
+    mocked_s3,
+    runner,
+):
     result = runner.invoke(main)
     assert result.exit_code == 0
-    assert "Logger 'root' configured with level=DEBUG" in caplog.text
+    assert "Logger 'root' configured with level=INFO" in caplog.text
     assert "Patronload config settings loaded for environment: test" in caplog.text
     assert "Running patronload process" in caplog.text
+    assert (
+        "'staff_2023-03-01_12.00.00.zip' uploaded to S3 bucket 'test-bucket'"
+        in caplog.text
+    )
+    assert (
+        "'student_2023-03-01_12.00.00.zip' uploaded to S3 bucket 'test-bucket'"
+        in caplog.text
+    )
+    s3_bucket_path_contents = mocked_s3.list_objects_v2(
+        Bucket="test-bucket", Prefix="patronload"
+    )["Contents"]
+    assert (
+        s3_bucket_path_contents[0]["Key"] == "patronload/staff_2023-03-01_12.00.00.zip"
+    )
+    assert (
+        s3_bucket_path_contents[1]["Key"]
+        == "patronload/student_2023-03-01_12.00.00.zip"
+    )
     assert "Total time to complete process" in caplog.text
+
+
+@freeze_time("2023-03-01 12:00:00")
+@patch("patronload.database.oracledb")
+def test_cli_duplicate_krb_name_remains_staff_patron(
```

Comment:
If I change the order of these lines ~55 in `cli.py` 
```python
"staff": {"fields": STAFF_FIELDS, "table": "LIBRARY_EMPLOYEE"},
"student": {"fields": STUDENT_FIELDS, "table": "LIBRARY_STUDENT"},
```
so that student records are processed first (as we discussed, someone might do in the future if they didn't know about staff needing to take precedence), then this test doesn't pass, which is good, but it fails for the 'wrong' reason. And I'm not sure it's clear from the test that staff need to take precedence. Maybe the solution is just a note in the doc string or a comment at line 55 in `cli.py`?

I'm leaving this is a non-blocking comment.

Reply:
Good call, I think the comment is the best approach